enhancements/cluster-logging/hypershift-audit-logs.md

```diff
@@ -0,0 +1,249 @@
+- Forward API server audit logs from the management cluster, as they are not available in the hosted cluster.
+- Replace the SD Splunk forwarder with the CLO which is more flexible and feature-rich.
+- Separate configuration so that service consumer can request management cluster forwarding, but the service provider has final control.
+
+### User Stories
+
+#### As a service consumer, I want to forward selected API audit logs to my chosen destinations
+
+- configure API audit policy and forwarding using guest cluster resource.
+- have the desired audit logs forwarded from the management cluster to my destination.
+
+#### As a service provider, I want to forward selected API audit logs to my chosen destinations
+- configure the SD internal audit policy
+- configure API audit policy and forwarding using name-spaced logging customer resources in the management cluster.
+  - for example: Red Hat Service Delivery team forwards audit logs to internal splunk destination.
+- independent of user configuration.
+
+### Goals
+
+- Multiple audit policies can be defined in a management cluster, independent of hosted cluster policies.
+- Multiple audit policies can be defined in a hosted cluster, independent of the management cluster policies.
+  See [api-audit-log-policy](/api-audit-log-policy.md) for how the CLO will support this.
+- Allow some forwarding in the _management_ cluster to be configured from the _hosted_ cluster.
+- Keep direct control of management cluster forwarding on the management cluster.
+  - hosted cluster has no direct access to control plane configuration.
+  - data plane resources can request forwarding by the management cluster.
+  - management cluster HLO automatically validates data-plane requests, and enables them if they are safe.
+
+### Non-Goals
+
+- This proposal only covers API audit logs. No change to application, infrastructure or node audit logs.
+- Management cluster is not _required_ to honour all possible forwarding configurations allowed by CLF. \
+  Whether or how to restricting management cluster forwarding is an operational decision.
+
+## Proposal
+
+### Workflow Description
+
+#### Hosted Cluster, data plane
+
+1. Deploy the CLO as usual.
+1. Create a `Policy.audit.k8s.io` resource with the desired policy
+   (see [api-audit-log-policy](/api-audit-log-policy.md))
+1. Create a `HypershiftLogForwarder` (HLF) resource with an `audit` input linked to the policy. \
+   The `HypershiftLogForwarder` API is a restricted subset of the `ClusterLogForwarder` API
+1. Create HLF `outputs` and `pipelines` to forward the resulting logs.
+1. Optionally create a `ClusterLogForwarder` to forward non-API logs as usual.
+
+#### Management Cluster
+
+1. Deploy the CLO, all name-spaces.
+1. Deploy the new Hypershift Logging Operator (HLO), watching all namespaces.
+1. Create a single `HypershiftLogWatcher` resource containing a template for HCP logging.
```

Comment:
I read "single" as "singleton instance".  This means we have a single template for forwarding configuration that must be shared across all HCP on that management cluster.  We need to make this namespace scoped.  Deployment of that resource is a concern for the service provider.  It provides flexibility at a HCP level.

Reply:
Fixed - instance per HCP as discussed.